pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -4,33 +4,77 @@ fixupOutputHooks+=(_doStrip)
 
 _doStrip() {
     if [ -z "$dontStrip" ]; then
-        stripDebugList=${stripDebugList:-lib lib32 lib64 libexec bin sbin}
-        if [ -n "$stripDebugList" ]; then
-            stripDirs "$stripDebugList" "${stripDebugFlags:--S}"
+        if [ -n "${stripDebugList:-*}" ]; then
+            stripDirs "${stripDebugList:-*}" "${stripDebugFlags:--S}"
         fi
 
-        stripAllList=${stripAllList:-}
-        if [ -n "$stripAllList" ]; then
-            stripDirs "$stripAllList" "${stripAllFlags:--s}"
+        if [ -n "${stripAllList:-}" ]; then
+            stripDirs "${stripAllList:-}" "${stripAllFlags:--s}"
         fi
     fi
 }
```

Comment:
Nitpick: in `_doStrip` I would change all `:-` to `-`, so that people can explicitly define those variables as empty without being overridden back to the default values. (Yes, one could define them as `" "`, but it seems nicer to allow empty string as well.)

Note that otherwise e.g. the first `if` will *always* fire as `*` is used whenever the expansion could be empty.